### ISSUE TITLE
Conan can't find MSVC compiler with 20230612.1 version of windows-2022

### ISSUE THREAD

**Issue by ssrobins** [actions/runner-images#7741]
### Description

I've been installing Conan in the GitHub Actions images for C++ package management. I use precise settings for the compiler version and update version so that whenever the `windows-2022` image gets an MSVC compiler update, it triggers a new package build and I always have packages built with the same compiler as my application.

I was using `compiler.update=5` up until `20230606.1` but in `20230612.1` when `compiler.update=5` no longer finds the package, I change it to `compiler.update=6` and get a ` No CMAKE_CXX_COMPILER could be found` error.

The error is not reproducible in my local install of Visual Studio Enterprise 2022 17.6.3 with `Microsoft.VisualStudio.Component.VC.Tools.x86.x64` and the Conan version has not changed so this leads me to believe it is a change in the MSVC installation specific to  the `windows-2022` image. Any change in how MSVC is installed that could be causing this?

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Image version is `20230612.1`

Here's the failed run with the simplified workflow:
https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034

with the original workflow:
https://github.com/ssrobins/conan-recipes/actions/runs/5286662748/jobs/9566717240

### Is it regression?

Yes, worked in `20230517.1` with compiler.update=5: https://github.com/ssrobins/conan-recipes/actions/runs/5195787580/jobs/9368846487

### Expected behavior

I expect the specified compiler.update=6 to allow CMake build to find the MSVC compiler as it has done for several compiler updates before and as it still does with my local install of VS2022.

### Actual behavior

It fails to find the MSVC compiler:
```
Run conan install --requires=fmt/9.1.0 --build=missing -s compiler=msvc -s compiler.version=193 -s compiler.update=6

======== Input profiles ========
Profile host:
[settings]
arch=x86_6[4](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:5)
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.update=6
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
fmt/9.1.0: Not found in local cache, looking in remotes...
fmt/9.1.0: Checking remote: conancenter
fmt/9.1.0: Downloaded recipe revision 44302d39c[5](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:6)a4bf7de8a39adc50bb45[6](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:7)8
Graph root
    cli
Requirements
    fmt/9.1.0#44302d39c5a4bf[7](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:8)de[8](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:9)a3[9](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:10)adc50bb4568 - Downloaded (conancenter)

======== Computing necessary packages ========
fmt/9.1.0: Checking 3 compatible configurations:
fmt/9.1.0: '894d4927b5c8e142[10](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:11)[11](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:12)16f1e038453292eaafa2': compiler.cppstd=17
fmt/9.1.0: 'e60e506c23c016790202021e83c36f2ce1bf03ee': compiler.cppstd=20
fmt/9.1.0: '4379a99de7fc102a6ef8ef0b4d09302d16b67958': compiler.cppstd=23
Requirements
    fmt/9.1.0#44302d39c5a4bf7de8a39adc50bb4568:91441c9c4a1b8563b5f9cbc27b2d349511143[12](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:13)3 - Build

======== Installing packages ========
fmt/9.1.0: Sources downloaded from 'conancenter'
fmt/9.1.0: Calling source() in C:\Users\runneradmin\.conan2\p\fmt7a98dde3dc37b\s\src
fmt/9.1.0: Downloading 1.3MB fmt-9.1.0.zip
fmt/9.1.0: Unzipping 4.5MB, this can take a while


-------- Installing package fmt/9.1.0 (1 of 1) --------
fmt/9.1.0: Building from source
fmt/9.1.0: Package fmt/9.1.0:9[14](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:15)41c9c4a1b8563b5f9cbc27b2d349511143123
fmt/9.1.0: Copying sources to build folder
fmt/9.1.0: Building your package in C:\Users\runneradmin\.conan2\p\b\fmt7789eb271e8e6\b
fmt/9.1.0: Calling generate()
fmt/9.1.0: Generators folder: C:\Users\runneradmin\.conan2\p\b\fmt7789eb271e8e6\b\build\generators
fmt/9.1.0: CMakeToolchain generated: conan_toolchain.cmake
fmt/9.1.0: CMakeToolchain generated: CMakePresets.json
fmt/9.1.0: CMakeToolchain generated: ..\..\src\CMakeUserPresets.json
fmt/9.1.0: Generating aggregated env files
fmt/9.1.0: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
fmt/9.1.0: Calling build()
fmt/9.1.0: apply_conandata_patches(): No patches defined in conandata
fmt/9.1.0: Running CMake.configure()
fmt/9.1.0: RUN: cmake -G "Visual Studio [17](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:18) [20](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:21)[22](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:23)" -DCMAKE_TOOLCHAIN_FILE="C:/Users/runneradmin/.conan2/p/b/fmt7789eb271e8e6/b/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/runneradmin/.conan2/p/b/fmt7789eb271e8e6/p" -DFMT_DOC="OFF" -DFMT_TEST="OFF" -DFMT_INSTALL="ON" -DFMT_LIB_DIR="lib" -DFMT_OS="ON" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\runneradmin\.conan2\p\b\fmt7789eb271e8e6\b\src"
-- CMake version: 3.26.4
-- Using Conan toolchain: C:/Users/runneradmin/.conan2/p/b/fmt7789eb271e8e6/b/build/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 14 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:66 (project):
  No CMAKE_CXX_COMPILER could be found.



-- Configuring incomplete, errors occurred!

fmt/9.1.0: ERROR: 
Package '91441c9c4a1b8563b5f9cbc27b2d3495111431[23](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:24)' build failed
fmt/9.1.0: WARN: Build folder C:\Users\runneradmin\.conan2\p\b\fmt7789eb[27](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:28)1e8e6\b\build
*********************************************************
Recipe 'fmt/9.1.0' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter check: https://conan.io/cci-v2.html
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: fmt/9.1.0: Error in build() method, line [94](https://github.com/ssrobins/actions-fiddling/actions/runs/5299640034/jobs/9592721846#step:4:95)
	cmake.configure()
	ConanException: Error 1 while executing
Error: Process completed with exit code 1.
1s
```

### Repro steps

I distilled this down to a simple set of repro steps in the form of GitHub Actions script:
https://github.com/ssrobins/actions-fiddling/tree/missing_compiler
```
name: CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_dispatch:

jobs:
  run:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [windows-2022]
    steps:
      - uses: actions/checkout@v3

      - name: Install Conan
        run: |
          pip3 install conan --upgrade
          conan profile detect

      - name: Conan install compiler.update=6
        run: conan install --requires=fmt/9.1.0 --build=missing -s compiler=msvc -s compiler.version=193 -s compiler.update=6
```

**Comment by ssrobins** [comment]
Thanks, here's a bit more information on what Conan is actually doing to detect the compiler:
https://github.com/conan-io/conan/issues/14111#issuecomment-1596166875

**Comment by ilia-shipitsin** [comment]
@ssrobins , do you know where I can find `conan_toolchain.cmake` ?

**Comment by ilia-shipitsin** [comment]
as for workloads, we install the following pieces

https://github.com/actions/runner-images/blob/main/images/win/toolsets/toolset-2022.json#L181-L305

**Comment by vpolikarpov-akvelon** [comment]
Hey, @ssrobins.
I assume that something is wrong with `Microsoft.VisualStudio.Component.VC`. Looks like version `14.36.17.6` may be broken. Could you check what version is in your local environment?

Also, could you check once more whether `conan install` command you provided works in your local environment in each of these cases:
- with `compiler.update=5`
- with `compiler.update=6`
- with no value for `compiler.update`

**Comment by ssrobins** [comment]
@ilia-shipitsin conan_toolchain.cmake is generated by the `conan install` command.

@vpolikarpov-akvelon I noticed that simply having 17.6.3 installed wasn't enough to install the 14.36 build tools on my machine. Once I installed the 14.36 components, I get a local repo with:

```
pip3 install conan --upgrade
conan profile detect
conan install --requires=fmt/9.1.0 --build=missing -s compiler=msvc -s compiler.version=193 -s compiler.update=6
```

It still works fine with `compiler.update=5` and no `compiler.update` setting.

Thanks for the investigation. https://github.com/actions/runner-images/pull/7756 should unblock me, let me know when it's live and I can verify.
